packages/sanity/src/_internal/cli/commands/backup/downloadBackupCommand.ts

```diff
@@ -0,0 +1,297 @@
+    // of 256 chars on path names and both of them can be quite long in some cases.
+    const tmpOutDir = path.join(tmpdir(), `backup-${Date.now()}-${randomKey(5)}`)
+
+    // Create required directories if they don't exist.
+    for (const dir of [
+      outDir,
+      tmpOutDir,
+      path.join(tmpOutDir, 'images'),
+      path.join(tmpOutDir, 'files'),
+    ]) {
+      if (!existsSync(dir)) {
+        mkdirSync(dir, {recursive: true})
+      }
+    }
+
+    debug('Writing to temporary directory %s', tmpOutDir)
+    const tmpOutDocumentsFile = path.join(tmpOutDir, 'data.ndjson')
+
+    // Handle concurrent writes to the same file using mutex.
+    const docOutStream = createWriteStream(tmpOutDocumentsFile, {flags: 'a'})
+    const docWriteMutex = new Mutex()
+
+    try {
+      const backupFileStream = new PaginatedGetBackupStream(
+        client,
+        opts.projectId,
+        opts.datasetName,
+        opts.backupId,
+        opts.token,
+      )
+
+      const files: File[] = []
+      let i = 0
+      for await (const file of backupFileStream) {
+        files.push(file)
+        i++
+        progressSpinner.set({
+          step: `Reading backup files...`,
+          update: true,
+          current: i,
+          total: backupFileStream.totalFiles,
+        })
+      }
+
+      let totalItemsDownloaded = 0
+      // This is dynamically imported because this module is ESM only and this file gets compiled to CJS at this time.
+      const {default: pMap} = await import('p-map')
+      await pMap(
+        files,
+        async (file: File) => {
+          if (file.type === 'file' || file.type === 'image') {
+            await downloadAsset(file.url, file.name, file.type, tmpOutDir)
+          } else {
+            const doc = await downloadDocument(file.url)
+            await docWriteMutex.runExclusive(() => {
+              docOutStream.write(`${doc}\n`)
+            })
+          }
+
+          totalItemsDownloaded += 1
+          progressSpinner.set({
+            step: `Downloading documents and assets...`,
+            update: true,
+            current: totalItemsDownloaded,
+            total: backupFileStream.totalFiles,
+          })
+        },
+        {concurrency: opts.concurrency},
+      )
+    } catch (error) {
+      progressSpinner.fail()
+      let msg = error.statusCode ? error.response.body.message : error.message
+      // If no message can be extracted, print the whole error.
+      if (msg === undefined) {
+        msg = String(error)
+      }
+      throw new Error(`Downloading dataset backup failed: ${msg}`)
+    }
+
+    progressSpinner.set({step: `Archiving files into a tarball...`, update: true})
+    try {
+      await archiveDir(tmpOutDir, outFilePath, (processedBytes: number) => {
+        progressSpinner.update({
+          step: `Archiving files into a tarball, ${humanFileSize(processedBytes)} bytes written...`,
```

Comment:
I remember that you showed me an `assets.json` but I don't see that created here. Is it needed?

Reply:
I think you missed this comment.

Reply:
From a quick glance at the export/import code, it seems like this file is used to track asset props for all the assets in one file and delete the actual asset documents from export. So, while importing, you would go over all the regular documents and then go over this asset.json file to import corresponding asset to CDN and its props to Content Lake.

In our case, we are already storing asset documents and asset files as they are. Upon import, they are being stored correctly in Content Lake and CDN. There may be a slight difference in these two variations, so going to look into this in-depth now.

Reply:
If it isn't needed then that's great! I'm just checking in on this point.